feat(Angular): apply changes in application styles at runtime with HMR

vchimev opened the pull request:
<!--
We, the rest of the NativeScript community, thank you for your
contribution! 
To help the rest of the community review your change, please follow the instructions in the template.
-->

Update demo and template.

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA].
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-dev-webpack/blob/master/CONTRIBUTING.md#testing-locally-by-running-e2e-tests
- [ ] Tests for the changes are included.

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Currently, a style change in `app.css` or `app.scss` is not accepted and causes application to restart.

## What is the new behavior?
<!-- Describe the changes. -->

The new behavior accepts changes in `app.css` or `app.scss` and uses the HMR context passed by `nativescript-dev-webpack` to determinate whether changes have been made to `app.css` or `app.scss`. If so, apply them at runtime and on back navigation.

Relates to https://github.com/NativeScript/NativeScript/pull/6665.
Implements the rest of https://github.com/NativeScript/NativeScript/issues/6645.

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

<!-- 
BREAKING CHANGES:


[Describe the impact of the changes here.]

Migration steps:
[Provide a migration path for existing applications.]
-->


[CLA]: http://www.nativescript.org/cla

vakrilov commented on bundle-config-loader.js:
```diff
@@ -2,38 +2,61 @@ module.exports = function (source) {
+                });
+            }
+
+            global.__hmrInitialSync = true; // needed to determine if we are performing initial sync
```
I think this one is not needed. 
It was used before in a old version of [hot-loader-helper](https://github.com/NativeScript/nativescript-dev-webpack/blob/172e49866299a47fd726e14af227465ca43ca48f/hot-loader-helper.js#L7).Not used anymore though.

sis0k0 commented:
Update the `main.ts` snippet here when this change is released - https://github.com/NativeScript/nativescript-angular/wiki/HMR.
Also, make sure to include migration instructions in the changelog.

vchimev commented:
Added [NativeScript 5.2 HMR with Angular](https://github.com/NativeScript/nativescript-angular/wiki/NativeScript-5.2-HMR-with-Angular) wiki page.
Added [migration instructions](https://github.com/NativeScript/nativescript-dev-webpack/blob/205d3e2f5ccfae2a8fa86a4e747909cf13526d41/CHANGELOG.md#migration) in the changelog.
